pages/expense.js

```diff
@@ -212,8 +222,25 @@ class ExpensePage extends React.Component {
     return sortBy([...comments, ...activities], 'createdAt');
   });
 
+  onSuccessMsgDismiss = () => {
+    // Replaces the route by the version without `createSuccess=true`
+    const { parentCollectiveSlug, collectiveSlug, legacyExpenseId } = this.props;
+    Router.replaceRoute(
+      `expense-v2`,
+      {
+        parentCollectiveSlug,
+        collectiveSlug,
+        collectiveType: parentCollectiveSlug && 'events',
+        ExpenseId: legacyExpenseId,
+      },
+      {
+        shallow: true, // Do not re-fetch data, do not loose state
```

Comment:
This one is a nice friend, it removes the `?createSuccess=true` from the route while preserving the state of the page (typed comment, editing expense, ...).